griptape/common/prompt_stack/prompt_stack.py

```diff
@@ -81,6 +78,7 @@ def __to_message_content(self, artifact: str | BaseArtifact) -> list[BaseMessage
                 return [ActionResultMessageContent(output, action=action)]
         elif isinstance(artifact, ListArtifact):
             processed_contents = [self.__to_message_content(artifact) for artifact in artifact.value]
+
             return [sub_content for processed_content in processed_contents for sub_content in processed_content]
         else:
-            raise ValueError(f"Unsupported artifact type: {type(artifact)}")
+            return [TextMessageContent(TextArtifact(artifact.to_text()))]
```

Comment:
Why this change in behavior?

Plus, you updated `TextMessageContent` to accept a `BaseArtifact` instead of `TextArtifact`, so do you need to convert it here at all?

Reply:
Originally it was to allow any Artifact (such as a custom user Artifact) to be passed without having to subclass ours. But maybe we should revert and just have users subclass.

Reply:
Actually, yeah it was to avoid handling Artifacts beyond the ones explicitly matched here. For instance, `ErrorArtifact` or `InfoArtifact`. I am going to revert the change to `text_message_content` though.